### ISSUE TITLE
Option to automatically calculate total size of files in a path

### ISSUE THREAD

**Issue by alphapapa** [tqdm/tqdm#504]
- [x] I have visited the [source website], and in particular
  read the [known issues]
- [x] I have searched through the [issue tracker] for duplicates
- [x] I have mentioned version numbers, operating system and
  environment, where applicable

Hi,

`tqdm` is very useful in `tar | gz` pipes when used like:

```
tar c $path | tqdm --unit B --bytes --unit_scale --total $(du -bs $path | cut -f1) | gzip $path
```

It would be nice if `tqdm` could automatically compute the total size of files in a directory path (including subdirectories) to avoid having to run `du` manually, something like:
```
tar c $path | tqdm --unit B --bytes --unit_scale --total-dir-size $path | gzip $path
```
Thanks for your work on tqdm!

**Comment by casperdcl** [comment]
hmm not sure about this. Seems like a lot of python code for something which the shell can do quite nicely.

**Comment by alphapapa** [comment]
> Seems like a lot of python code for something which the shell can do quite nicely.

That's true.  The reason I suggested it is that it seems like a fairly straightforward thing to do, but to do it with shell commands likely means consulting the `du` man page for the right options, verifying its output, then testing piping it through `cut` (e.g. I expected to use `cut -d ' ' -f1` like I do for most other commands, but the separator is a tab, so that didn't work), and then finally running the whole composite command.  If `tqdm` supported it natively, it would turn a 4-5 step process into a 0-step process.  :)

**Comment by StanczakDominik** [comment]
It might be handy but it's probably way out of scope, and can instead be handled by the user via a bash alias. Perhaps it would be better to close this one?

**Comment by casperdcl** [comment]
Hah I rediscovered the "4-5" step process when making these slides https://tqdm.github.io/PyData2019/slides.html#/5/4 and never thought about it - it's just bash programming.

My usual POSIX-based philosophy is "do one thing well" but since Python's is "do everything with pseudocode" I'm a bit conflicted.

I don't know. Can someone post a cross-platform python code snippet `def bytes_size(file_or_path)`?
